File: sol/1574/1574.cpp
struct Solution {
    int findLengthOfShortestSubarray(vector<int>& arr) {
        int n = arr.size(), prefix = 0;
        while (prefix < n - 1 && arr[prefix] <= arr[prefix + 1]){
            ++prefix;
        }
        int i = prefix, res = prefix + 1, prev = arr[n - 1] + 1;
        for (int j = n - 1; j > prefix; --j){
            if (arr[j] > prev){
                break;
            }
            prev = arr[j];
            while (i >= 0 && arr[i] > arr[j]){
                --i;
            }
            res = max(res, n + 1 - j + i);
        }
        return n - res;
    }
};

File: sol/1574/1574.py
class Solution:
    def findLengthOfShortestSubarray(self, arr: List[int]) -> int:
        n = len(arr)
        prefix = 0
        while prefix < n - 1 and arr[prefix] <= arr[prefix + 1]:
            prefix += 1
        i = prefix
        res = prefix + 1
        prev = arr[n - 1] + 1
        for j in range(n - 1, prefix, -1):
            if arr[j] > prev:
                break
            prev = arr[j]
            while i >= 0 and arr[i] > arr[j]:
                i -= 1
            res = max(res, n + 1 + i - j)
        return n - res
